tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -364,6 +368,168 @@ public void testRsaWithoutCrtParams() throws Exception {
+                continue;
+            }
+            final int trailer = ii;
+            assertThrows(IllegalArgumentException.class, () -> signature.setParameter(
+                new PSSParameterSpec("SHA-1", "MGF1", MGF1ParameterSpec.SHA1, 20, trailer)
+            ));
+        }
+    }
+
+    @Test
+    public void testRsaPSSDefaultsToSha1() throws Exception {
+        final KeyPairGenerator kg = KeyPairGenerator.getInstance("RSA", AmazonCorrettoCryptoProvider.INSTANCE);
+        kg.initialize(2048);
+        final KeyPair pair = kg.generateKeyPair();
+        Signature signature;
+        PSSParameterSpec spec;
+
+        signature = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        signature.initSign(pair.getPrivate());
+        spec = signature.getParameters().getParameterSpec(PSSParameterSpec.class);
+        assertEquals("SHA-1", spec.getDigestAlgorithm());
+        assertEquals("SHA-1", ((MGF1ParameterSpec) spec.getMGFParameters()).getDigestAlgorithm());
+
+        signature = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        signature.initVerify(pair.getPublic());
+        spec = signature.getParameters().getParameterSpec(PSSParameterSpec.class);
+        assertEquals("SHA-1", spec.getDigestAlgorithm());
+        assertEquals("SHA-1", ((MGF1ParameterSpec) spec.getMGFParameters()).getDigestAlgorithm());
+
+        // setting null params OK, shouldn't change from default
+        signature = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        signature.setParameter(null);
+        signature.initSign(pair.getPrivate());
+        spec = signature.getParameters().getParameterSpec(PSSParameterSpec.class);
+        assertEquals("SHA-1", spec.getDigestAlgorithm());
+        assertEquals("SHA-1", ((MGF1ParameterSpec) spec.getMGFParameters()).getDigestAlgorithm());
+    }
+
+    @Test
+    void testRsaPSSTryUpdateParamDuringBuffer() throws Exception {
+        final KeyPairGenerator kg = KeyPairGenerator.getInstance("RSA", AmazonCorrettoCryptoProvider.INSTANCE);
+        kg.initialize(2048);
+        final KeyPair pair = kg.generateKeyPair();
+        final Signature signer = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        final Signature verifier = Signature.getInstance("RSASSA-PSS", NATIVE_PROVIDER);
+        final PSSParameterSpec spec1 = new PSSParameterSpec("SHA-224", "MGF1", MGF1ParameterSpec.SHA224, 224/8, 1);
+        final PSSParameterSpec spec2 = new PSSParameterSpec("SHA-256", "MGF1", MGF1ParameterSpec.SHA256, 256/8, 1);
+
+        // Initialize signatures and set non-default PSS parameters
+        signer.initSign(pair.getPrivate());
+        verifier.initVerify(pair.getPublic());
+        signer.setParameter(spec1);
+        verifier.setParameter(spec1);
+
+        // Assert that set values persisted
+        assertEquals(
+            spec1.getDigestAlgorithm(),
+            signer.getParameters().getParameterSpec(PSSParameterSpec.class).getDigestAlgorithm()
+        );
```

Comment:
You might find it helpful to create a new `assertPssParamsEquals()` method.

Reply:
yeah, might as well.